SampleApp/ViewController.swift

```diff
@@ -7,62 +7,69 @@ import RxSwift
-  
-  override func viewDidLoad() {
-    super.viewDidLoad()
-    
-    socket.rx.response
-      .subscribe(onNext: { [weak self] (response: WebSocketEvent) in
-        
-        guard let `self` = self else {
-          return
-        }
-        
-        switch response {
-        case .connected:
-          self.logTextView.text = self.logTextView.text + "Connected\n"
-        case .disconnected(let error):
-          self.logTextView.text = self.logTextView.text + "Disconnected with optional error: \(error) \n "
-        case .message(let msg):
-          self.logTextView.text = self.logTextView.text + "RESPONSE (Message): \(msg) \n"
-        case .data(let data):
-          self.logTextView.text = self.logTextView.text + "RESPONSE (Data): \(data) \n"
-        case .pong:
-          self.logTextView.text = self.logTextView.text + "RESPONSE (Pong)"
-        }
-      }).addDisposableTo(disposeBag)
-    
-    socket.connect()
-  }
 
-  @IBAction func sendButtonAction(sender: UIButton) {
-    if let text = inputTextField.text, !text.isEmpty {
-      sendMessage(message: text)
+    @IBOutlet weak var inputTextField: UITextField!
+    @IBOutlet weak var sendButton: UIButton!
+    @IBOutlet weak var logTextView: UITextView!
+    @IBOutlet weak var pongButton: UIBarButtonItem!
+
+    let disposeBag = DisposeBag()
+    let socket = WebSocket(url: URL(string: "wss://echo.websocket.org")!)
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        sendButton.rx.tap
+                .subscribe(onNext: { [unowned self] in
+                    guard let text = self.inputTextField.text, !text.isEmpty else {
+                        return
+                    }
+                    self.sendMessage(message: text)
+                }).addDisposableTo(disposeBag)
+
+        pongButton.rx.tap.subscribe(onNext: { [unowned self] in
+                    self.socket.write(ping: Data())
+                }).addDisposableTo(disposeBag)
+
+        socket.rx.response
+                .subscribe(onNext: { [weak self] (response: WebSocketEvent) in
+
+                    guard let `self` = self else {
+                        return
+                    }
+
+                    switch response {
+                    case .connected:
+                        self.logTextView.text = self.logTextView.text + "Connected\n"
+                    case .disconnected(let error):
+                        self.logTextView.text = self.logTextView.text + "Disconnected with optional error: \(error) \n "
+                    case .message(let msg):
+                        self.logTextView.text = self.logTextView.text + "RESPONSE (Message): \(msg) \n"
+                    case .data(let data):
+                        self.logTextView.text = self.logTextView.text + "RESPONSE (Data): \(data) \n"
+                    case .pong:
+                        self.logTextView.text = self.logTextView.text + "RESPONSE (Pong)"
+                    }
```

Comment:
Just a gotcha here but I would map the response to a String and then do asDriver and `driver(self.logTextView.rx.text)` , better separation of concern and less repeating yourself.

Unless we don't have RxCocoa in this project in which case you can just map to String and then change the text field in one place.